src/test/java/com/yahoo/bullet/bql/BulletQueryBuilderTest.java

```diff
@@ -5,768 +5,1905 @@
+        Assert.assertEquals(query.getAggregation().getFields().size(), 2);
+        Assert.assertEquals(query.getAggregation().getFields().get("abc"), "abc");
+        Assert.assertEquals(query.getAggregation().getFields().get("def"), "def");
+        Assert.assertNull(query.getAggregation().getAttributes());
+        Assert.assertNull(query.getAggregation().getSize());
+        Assert.assertNull(query.getPostAggregations());
+    }
+
+    @Test
+    public void testDistinctWithAlias() {
+        build("SELECT DISTINCT abc AS one, def AS two FROM STREAM()");
+        Assert.assertNull(query.getProjection());
+        Assert.assertEquals(query.getAggregation().getType(), Aggregation.Type.GROUP);
+        Assert.assertEquals(query.getAggregation().getFields().size(), 2);
+        Assert.assertEquals(query.getAggregation().getFields().get("abc"), "one");
+        Assert.assertEquals(query.getAggregation().getFields().get("def"), "two");
+        Assert.assertNull(query.getAggregation().getAttributes());
+        Assert.assertNull(query.getAggregation().getSize());
+        Assert.assertNull(query.getPostAggregations());
+    }
+
+    @Test
+    public void testDistinctWithComputation() {
+        build("SELECT DISTINCT abc + 5 FROM STREAM()");
+        Assert.assertEquals(query.getProjection().getFields().size(), 1);
+        Assert.assertEquals(query.getProjection().getFields().get(0), new Field("abc + 5", new BinaryExpression(new FieldExpression("abc"),
+                                                                                                                new ValueExpression(5),
+                                                                                                                Operation.ADD)));
+        Assert.assertEquals(query.getAggregation().getType(), Aggregation.Type.GROUP);
+        Assert.assertEquals(query.getAggregation().getFields().size(), 1);
+        Assert.assertEquals(query.getAggregation().getFields().get("abc + 5"), "abc + 5");
+        Assert.assertNull(query.getAggregation().getAttributes());
+        Assert.assertNull(query.getAggregation().getSize());
+        Assert.assertNull(query.getPostAggregations());
+    }
+
+    @Test
+    public void testDistinctWithOrderBy() {
+        build("SELECT DISTINCT abc FROM STREAM() ORDER BY abc");
+        Assert.assertEquals(query.getPostAggregations().size(), 1);
+
+        OrderBy orderBy = (OrderBy) query.getPostAggregations().get(0);
+
+        Assert.assertEquals(orderBy.getFields().size(), 1);
+        Assert.assertEquals(orderBy.getFields().get(0).getField(), "abc");
+    }
+
+    @Test
+    public void testDistinctWithInvalidOrderByField() {
+        build("SELECT DISTINCT abc FROM STREAM() ORDER BY def");
+        Assert.assertEquals(errors.get(0).getError(), "ORDER BY contains a non-existent field: def");
+        Assert.assertEquals(errors.size(), 1);
+    }
+
+    @Test
+    public void testDistinctWithOrderByComputation() {
```

Comment:
At least let's push each category of test to its own file?

Reply:
Uh I'll try